base/multidimensional.jl

```diff
@@ -207,11 +207,11 @@ function _unsafe_getindex(::LinearIndexing, src::AbstractArray, I::AbstractArray
     D = eachindex(dest)
     Ds = start(D)
     s = 0
-    for b in eachindex(I)
+    @inbounds for b in eachindex(I)
```

Comment:
Oooh, my understanding here is wrong.  I'll restrict these `@inbounds` cases back to the status quo for this PR… and we can work on how this works later.